openupgradelib/openupgrade_merge_records.py

```diff
@@ -421,6 +425,23 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
             if op == 'merge':
                 if not getattr(target_record, field.name) and _list:
                     vals[field.name] = _list[0]
+        elif field.type == 'many2one_reference':
+            if field.model_field not in model._fields:
+                continue
+            model_field = [x for x in fields if x.name == field.model_field][0]
+            op = op or 'merge'
+            if op == 'merge':
+                if field.model_field not in field_spec:
```

Comment:
It shouldn't matter if it's on field_spec or not, we should perform the operation IMO. In fact, we must remove the entry from `field_spec` if exists.